### PR TITLE
Segregate XRP

### PR DESCRIPTION
**Pull request by phaazon**
# Content

This PR contains the _whole_ XRP modularization. Currently, it doesn’t contain the tests, which will come in a future PR.

# Mandatory picture of animal (hello @onyb 👋 )

### CODE REVIEW

File: ledger-core-ripple/inc/ripple/RippleLikeOperation.hpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            );
+            std::shared_ptr<api::RippleLikeTransaction> getTransaction() override;
+
+            void refreshUid() override;
+
+        private:
+            std::shared_ptr<api::RippleLikeTransaction> _transaction;
+        };
+
+        class RippleLikeOperationQuery : public OperationQuery<RippleLikeOperation> {
```
Comment (alekece) [inline ledger-core-ripple/inc/ripple/RippleLikeOperation.hpp]: I suggest to hold only one class by file - avoid potential confusion about which header should be included.

File: ledger-core-ripple/inc/ripple/explorers/NodeRippleLikeBlockchainExplorer.hpp [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                _document.Accept(writer);
+                return buffer.GetString();
+            };
+
+        private:
+            rapidjson::Document _document;
+            rapidjson::Value _params;
+        };
+
+        class NodeRippleLikeBlockchainExplorer : public RippleLikeBlockchainExplorer,
```
Comment (alekece) [inline ledger-core-ripple/inc/ripple/explorers/NodeRippleLikeBlockchainExplorer.hpp]: Same suggestion here - one class/struct by header

File: ledger-core-ripple/inc/ripple/synchronizers/RippleLikeAccountSynchronizer.hpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+#pragma once
+
+#include <core/synchronizers/AbstractAccountSynchronizer.hpp>
+
+namespace ledger {
+    namespace core {
+        class RippleLikeAccount;
+
+        class RippleLikeAccountSynchronizer : public AbstractAccountSynchronizer<RippleLikeAccount> {
```
Comment (alekece) [inline ledger-core-ripple/inc/ripple/synchronizers/RippleLikeAccountSynchronizer.hpp]: An `using` should be enough here ?

File: ledger-core-ripple/inc/ripple/transaction_builders/RippleLikeTransactionBuilder.hpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+#include <core/async/Future.hpp>
+#include <core/math/BigInt.hpp>
+#include <core/wallet/Amount.hpp>
+
+#include <ripple/api/RippleLikeTransactionBuilder.hpp>
+#include <ripple/explorers/RippleLikeBlockchainExplorer.hpp>
+
+namespace ledger {
+    namespace core {
+        struct RippleLikeTransactionBuildRequest {
```
Comment (alekece) [inline ledger-core-ripple/inc/ripple/transaction_builders/RippleLikeTransactionBuilder.hpp]: Same suggestion here - one class/struct by header

File: ledger-core-ripple/src/ripple/RippleLikeCurrencies.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+#include <ripple/RippleLikeCurrencies.hpp>
+#include <ripple/RippleNetworks.hpp>
+
+namespace ledger {
+    namespace core {
+        namespace currencies {
+            api::Currency const RIPPLE =
+                CurrencyBuilder("ripple")
+                .bip44(144)
```
Comment (alekece) [inline ledger-core-ripple/src/ripple/RippleLikeCurrencies.cpp]: Maybe you can reuse the identifier declared in `Migration.hpp` ?

File: ledger-core/inc/core/LibCoreExport.hpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#pragma once
```
Comment (alekece) [inline ledger-core/inc/core/LibCoreExport.hpp]: May a dedicated - and small - PR for this file be better ?

File: ledger-core/inc/core/operation/Operation.hpp [hunk 7]
```diff
@@ -71,6 +71,8 @@ namespace ledger {
 
             api::Currency getCurrency() override;
 
+            bool isComplete() override;
```
Comment (alekece) [inline ledger-core/inc/core/operation/Operation.hpp]: ditto

File: ledger-core/inc/core/utils/DerivationScheme.hpp [hunk 8]
```diff
@@ -59,7 +58,7 @@ namespace ledger {
             DerivationScheme(const std::vector<DerivationSchemeNode>& nodes);
             DerivationScheme(const DerivationScheme& cpy);
             DerivationScheme getSchemeFrom(DerivationSchemeLevel level);
-            DerivationScheme getSchemeTo(DerivationSchemeLevel level);
+            DerivationScheme getSchemeTo(DerivationSchemeLevel level) const;
```
Comment (alekece) [inline ledger-core/inc/core/utils/DerivationScheme.hpp]: ditto

File: ledger-core-ripple/inc/ripple/explorers/NodeRippleLikeBlockchainExplorer.hpp (hunk 2 above)
Comment (phaazon) [reply ledger-core-ripple/inc/ripple/explorers/NodeRippleLikeBlockchainExplorer.hpp]: Nope, not here. I’ll move the `NodeRippleLikeBodyRequest` directly into the C++ implementation instead as it’s a detail of implementation.

**Pull request merged**